lib/livebook_web/live/path_select_component.ex

```diff
@@ -50,9 +50,23 @@ defmodule LivebookWeb.PathSelectComponent do
         <% end %>
       </div>
       <div class="flex-grow -m-1 p-1 overflow-y-auto tiny-scrollbar">
+        <div class="
+          grid grid-cols-4 gap-2
+          <%= if Enum.any?(list_files(@path, @extnames, @running_paths), fn file -> file.is_filtrate == false end) do%>
+            border-dashed border-b border-gray-200
+          <% end %>
+        ">
+          <%= for file <- list_files(@path, @extnames, @running_paths) do %>
+            <%= if file.is_filtrate do %>
+              <%= render_file(file, @phx_target) %>
+            <% end %>
+          <% end %>
+        </div>
         <div class="grid grid-cols-4 gap-2">
-          <%= for file <- list_matching_files(@path, @extnames, @running_paths) do %>
-            <%= render_file(file, @phx_target) %>
+          <%= for file <- list_files(@path, @extnames, @running_paths) do %>
+            <%= if not file.is_filtrate do %>
+              <%= render_file(file, @phx_target) %>
+            <% end %>
           <% end %>
```

Comment:
Implementation-wise, I think you could simplify this by introducing `file_groups` function that returns a list with file lists to show. You can then render each group in a separate container and use something like `divide-x divide-dashed` on the parent. Then in `file_groups` you could simply partition the files based on whether they have highlight and return that as two groups, and if no file matches it would just return a single group :>